jax/scipy/linalg.py

```diff
@@ -238,18 +237,28 @@ def triu(m, k=0):
 _expm_description = textwrap.dedent("""
 In addition to the original NumPy argument(s) listed below,
 also supports the optional boolean argument ``upper_triangular``
-to specify whether the ``A`` matrix is upper triangular.
+to specify whether the ``A`` matrix is upper triangular, and the optional
+argument ``max_n_squarings`` to specify the max number of squarings allowed
+in the scaling-and-squaring approximation method. Return nan if the actual
+number of squarings required is more than ``max_n_squarings``.
 """)
 
 @_wraps(scipy.linalg.expm, lax_description=_expm_description)
-def expm(A, *, upper_triangular=False):
-  return _expm(A, upper_triangular)
+def expm(A, *, upper_triangular=False, max_n_squarings=16):
+  return _expm(A, upper_triangular, max_n_squarings)
 
-@partial(api.custom_jvp, nondiff_argnums=(1,))
-def _expm(A, upper_triangular):
+def _expm(A, upper_triangular, max_n_squarings):
   P, Q, n_squarings = _calc_P_Q(A)
-  R = _solve_P_Q(P, Q, upper_triangular)
-  R = _squaring(R, n_squarings)
+
+  def _nan(_):
+    return jnp.zeros(A.shape, dtype=A.dtype) * jnp.nan
```

Comment:
Can you use `jnp.full_like` here?
```suggestion
    return jnp.full_like(A, jnp.nan)
```

Reply:
Done